libfwupd/fwupd-self-test.c

```diff
@@ -470,6 +470,210 @@ fwupd_device_filter_func(void)
 					       FWUPD_DEVICE_FLAG_ANOTHER_WRITE_REQUIRED));
 }
 
+static void
+fwupd_client_api_undefined_setter(void)
+{
+#if GLIB_CHECK_VERSION(2, 74, 0)
+	if (g_test_subprocess()) {
+		g_autoptr(FwupdClient) client = fwupd_client_new();
+		GValue value_bool = G_VALUE_INIT;
+
+		g_value_init(&value_bool, G_TYPE_BOOLEAN);
+		g_object_set_property(G_OBJECT(client), "battery-adapter", &value_bool);
+		g_value_unset(&value_bool);
+
+		return;
+	} else {
+		g_test_trap_subprocess(NULL, 0, G_TEST_SUBPROCESS_DEFAULT);
+		g_test_trap_assert_failed();
+		g_test_trap_assert_stderr(
+		    "*GLib-GObject-CRITICAL*has no property named 'battery-adapter'*");
+	}
+#else
+	g_test_skip("Trap handling requires glib 2.74");
+#endif
+}
+
+static void
+fwupd_client_api_undefined_getter(void)
+{
+#if GLIB_CHECK_VERSION(2, 74, 0)
+	if (g_test_subprocess()) {
+		g_autoptr(FwupdClient) client = fwupd_client_new();
+		GValue value_bool = G_VALUE_INIT;
+
+		g_value_init(&value_bool, G_TYPE_BOOLEAN);
+		g_object_get_property(G_OBJECT(client), "battery-adapter", &value_bool);
+		g_value_unset(&value_bool);
+
+		return;
+	} else {
+		g_test_trap_subprocess(NULL, 0, G_TEST_SUBPROCESS_DEFAULT);
+		g_test_trap_assert_failed();
+		g_test_trap_assert_stderr(
+		    "*GLib-GObject-CRITICAL*has no property named 'battery-adapter'*");
+	}
+#else
+	g_test_skip("Trap handling requires glib 2.74");
+#endif
+}
+
+static void
+fwupd_client_api_ro_props(void)
+{
+#if GLIB_CHECK_VERSION(2, 74, 0)
+	const gchar *prop;
+	const gchar *props[] = {"daemon-version", "tainted", "interactive", "only-trusted", NULL};
+
+	for (prop = props[0]; *prop; prop++) {
```

Comment:
I think for `(guint i = 0; i < G_N_ELEMENTS(props); i++)` (and not having a NULL) is probably easier to read.